src/index.js

```diff
@@ -14,10 +14,21 @@ const areEqual = (prevProps, nextProps) => {
   );
 };
 
-const Nouislider = props => {
+const Nouislider = forwardRef((props, ref) => {
   const [slider, setSlider] = useState(null);
   const sliderContainer = React.createRef();
 
+  useLayoutEffect(() => {
```

Comment:
Are there specific reasons for **useLayoutEffect** instead of **useEffect**?

Reply:
With my use case, I don't see any difference. What I do is to append a child:
```jsx
componentDidMount() {
  this.filler = document.createElement('div');
  this.filler.className = 'noUi-custom-filler';
  this.ref.current.appendChild(this.filler);
}

componentWillUnmount() {
  this.ref.current.removeChild(this.filler);
}
```

I am not sure if there won't be any flickering observed in other use cases, I am referring to [this](https://kentcdodds.com/blog/useeffect-vs-uselayouteffect) blog post by Kent Dodds.